pkg/asset/installconfig/azure/client.go

```diff
@@ -286,11 +286,15 @@ func (c *Client) GetVMCapabilities(ctx context.Context, instanceType, region str
 	if typeMeta == nil {
 		return nil, fmt.Errorf("not found in region %s", region)
 	}
-
+	if typeMeta.Family == nil {
+		return nil, fmt.Errorf("error getting resource family")
+	}
 	capabilities := make(map[string]string)
 	for _, capability := range *typeMeta.Capabilities {
 		capabilities[to.String(capability.Name)] = to.String(capability.Value)
 	}
+	capabilities["Family"] = to.String(typeMeta.Family)
```

Comment:
Why stuff family into capabilities? Capabilities corresponds to the Azure API and it seems counterintuitive to break with that model here. We see a use case here where we need the instance metadata and it seems likely that there will be future needs so I think it makes sense to expose it somehow

Reply:
I've rearranged the code so that there is now a call to get the resource family.